### ISSUE TITLE
[BUG] Mind Control combo point interaction

### ISSUE THREAD

**Issue by JamminL** [ClassicWoWCommunity/cata-classic-bugs#613]
#### 1.1. Description
Needs research still but seems like if you have combo points on a target and that target gets Mind Controlled, if you never untarget him, then you will not be able to see your combo points.

This is probably because combo points update in certain situations, but it's not updating from friendly->enemy events, which should.
  #### 1.2. How to Reproduce
           1. Get combo points on a target
           2. Have someone else Mind Control that target
           3. Keep the person targeted until MC is over
           4. Notice how your combo points no longer appear
  #### 1.3  Source Material
VIDEO OF BUG HAPPENING:
https://streamable.com/5049ip
EXAMPLE of when it works properly:
https://streamable.com/lezic9
#### 2.1. Description
You want combo points to always be visible in the UI.
  #### 2.2 Source Material
-

**Comment by XyzKangUI** [comment]
https://wowpedia.fandom.com/wiki/API_GetComboPoints

"GetComboPoints will return 0 if target is friendly or not found" -> This API logic is the reason that ComboFrame_Update function will hide the combo points when a target change / energy update event occurs.

> if you never untarget him, then you will not be able to see your combo points.

This is not entirely true. [Here ](https://github.com/Ketho/wow-ui-source-wrath/blob/3eaa29bfc00a05362e4cf30921a5c0792fc47519/Interface/FrameXML/ComboFrame.lua#L24)you can see the events when combo points update, one of them is while your energy is updating every decisecond. In the 'bug' video the rogue has capped his energy, so there are no events being fired (regardless of the target being friendly). 

This leaves the rogue with only one event, which is 'PLAYER_TARGET_CHANGED' to have his combo points re-appear on his current target. This is what you see in the 2nd video, where he swaps from paladin to warrior (who was not mind controlled at that point).

In conclusion, the rogue needs to spend his energy or change targets for his combo points to re-appear after MC.

I do have a workaround, which I posted on the [Arena Addicts discord](https://discord.gg/xNC7GF34dx)

**Comment by JamminL** [comment]
Archiving old non-relevant issues in preparation for MOP. If I made a mistake please comment on it.
